cpp/src/arrow/acero/hash_aggregate_test.cc

```diff
@@ -592,6 +592,11 @@ void TestSegments(std::unique_ptr<RowSegmenter>& segmenter, const ExecSpan& batc
     ASSERT_EQ(expected_segment, segment);
     offset = segment.offset + segment.length;
   }
+  // Assert next is the last (empty) segment.
+  ASSERT_OK_AND_ASSIGN(auto segment, segmenter->GetNextSegment(batch, offset));
+  ASSERT_TRUE(segment.offset >= batch.length);
+  ASSERT_TRUE(segment.is_open);
```

Comment:
Should something be checked about `segment.length` here? Or is undetermined?

Reply:
The last `segment.length` is supposed to be `0`. Updated.